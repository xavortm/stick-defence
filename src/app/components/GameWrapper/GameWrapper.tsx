import React, { useContext, useEffect } from 'react';
import styled from 'styled-components';

import { GameProvider, GameContext } from '../../context/store';
import Scene from '../Scene/Scene';
import Dashboard from '../Dashboard/Dashboard';
import Welcome from '../Welcome/Welcome';

import internalConfig from '../../gameConfig/internalConf';

const Game = styled.div`
  background: white;
  width: ${internalConfig.gameWindowWidth + 'px'};
  height: ${internalConfig.gameWindowHeight + 'px'};
  position: relative;
`;

export default function GameWrapper(): JSX.Element {
  return (
    <Game>
      <GameProvider>
        {/* Only visible when the game starts (or on page refresh) */}
        <Welcome />

        <Dashboard />
        <Scene />
      </GameProvider>
    </Game>
  );
}
