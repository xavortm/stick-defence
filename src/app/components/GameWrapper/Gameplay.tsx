import React, { useContext } from 'react';
import { GameContext } from 'app/context/store';
import Scene from '../Scene/Scene';
import Dashboard from '../Dashboard/Dashboard';
import Completed from '../GameEnd/Completed';
import Lost from '../GameEnd/Lost';

export default function Gameplay() {
  const { state } = useContext(GameContext);

  return state.gameplay.allWaves.length > 0 ? (
    <>
      <Dashboard />
      <Scene />

      {/* When the end game is reached, open the screen below. */}
      <Completed />
      <Lost />
    </>
  ) : null;
}
