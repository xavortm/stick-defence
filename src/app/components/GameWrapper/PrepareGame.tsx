import { useEffect, useContext } from 'react';
import { wavesSetup } from 'app/gameConfig/waves';
import { GameContext } from '../../context/store';
const PrepareGame = () => {
  const { dispatch } = useContext(GameContext);

  useEffect(() => {
    // const randomizedArray = [];

    // Prepare the wave for this game run (randomize stuff);
    // @todo - for some reason this doesn't dispatch to the reduced ........
    dispatch({ type: 'PREPARE_WAVES', payload: wavesSetup });
  }, [dispatch]);

  return null;
};

export default PrepareGame;
