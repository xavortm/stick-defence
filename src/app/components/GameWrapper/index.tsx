import React from 'react';
import styled from 'styled-components';

import { GameProvider } from '../../context/store';
import Scene from '../Scene/Scene';
import Dashboard from '../Dashboard/Dashboard';
import Welcome from '../Welcome/Welcome';
import Completed from '../GameEnd/Completed';
import Lost from '../GameEnd/Lost';

import internalConfig from '../../gameConfig/internalConf';

const Game = styled.div`
  background: white;
  width: ${internalConfig.gameWindowWidth + 'px'};
  height: ${internalConfig.gameWindowHeight + 'px'};
  position: relative;
`;

export default function GameWrapper(): JSX.Element {
  return (
    <Game>
      <GameProvider>
        {/* Only visible when the game starts (or on page refresh) */}
        <Welcome />

        <Dashboard />
        <Scene />

        {/* When the end game is reached, open the screen below. */}
        <Completed />
        <Lost />
      </GameProvider>
    </Game>
  );
}
