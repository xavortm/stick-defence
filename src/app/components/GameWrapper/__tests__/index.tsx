import * as React from 'react';
import { createRenderer } from 'react-test-renderer/shallow';

import GameWrapper from '../index';

const renderer = createRenderer();

describe('<GameWrapper />', () => {
  it('should render and match the snapshot', () => {
    renderer.render(<GameWrapper />);
    const renderedOutput = renderer.getRenderOutput();
    expect(renderedOutput).toMatchSnapshot();
  });
});
