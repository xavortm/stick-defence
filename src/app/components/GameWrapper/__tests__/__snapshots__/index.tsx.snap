// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GameWrapper /> should render and match the snapshot 1`] = `
<styled.div>
  <GameProvider>
    <Welcome />
    <Dashboard />
    <Scene />
    <Completed />
    <Lost />
  </GameProvider>
</styled.div>
`;
